### PR TITLE
HRCPP-204 Removed MurmurHash2 algorithm and ConsistentHashV1

### PR DESCRIPTION
**Pull request by rigazilla**
Removed murmurhash2 function and ConsistentHashV1 implementation

### CODE REVIEW

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 297](http://ci.infinispan.org/viewLog.html?buildId=32858&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for ff45f61f2210c9f5b434ce44c1d2a1d0a22dd8bc
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:37

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 298](http://ci.infinispan.org/viewLog.html?buildId=32859&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for ff45f61f2210c9f5b434ce44c1d2a1d0a22dd8bc
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:08

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 299](http://ci.infinispan.org/viewLog.html?buildId=32864&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for ff45f61f2210c9f5b434ce44c1d2a1d0a22dd8bc
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:39

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 300](http://ci.infinispan.org/viewLog.html?buildId=32865&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for ff45f61f2210c9f5b434ce44c1d2a1d0a22dd8bc
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:07:51

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 301](http://ci.infinispan.org/viewLog.html?buildId=32866&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for ff45f61f2210c9f5b434ce44c1d2a1d0a22dd8bc
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:07:55

File: src/hotrod/impl/consistenthash/ConsistentHashFactory.cpp [hunk 1]
```diff
@@ -14,9 +13,6 @@ using infinispan::hotrod::protocol::HotRodConstants;
 std::shared_ptr<ConsistentHash> ConsistentHashFactory::newConsistentHash(uint8_t version) {
     std::shared_ptr<ConsistentHash> result;
     switch (version) {
-        case HotRodConstants::CONSISTENT_HASH_V1:
```
Comment (tristantarrant) [inline src/hotrod/impl/consistenthash/ConsistentHashFactory.cpp]: Actually, better throw an "unsupported" exception here.

Comment (rigazilla) [inline src/hotrod/impl/consistenthash/ConsistentHashFactory.cpp]: then is better to add the throw in the default to get the same behaviour against unexpected "version". Agree?

Comment (tristantarrant) [inline src/hotrod/impl/consistenthash/ConsistentHashFactory.cpp]: +1

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 302](http://ci.infinispan.org/viewLog.html?buildId=32873&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 9f48bded57500405e54ec0644b943702c56316b2
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:24

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 303](http://ci.infinispan.org/viewLog.html?buildId=32874&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 9f48bded57500405e54ec0644b943702c56316b2
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:47

**Comment by tristantarrant** [comment]
I pushed only the commits related to $SUBJECT. For the typedef stuff, please use a new PR.

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 305](http://ci.infinispan.org/viewLog.html?buildId=32896&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 9f48bded57500405e54ec0644b943702c56316b2
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:00

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 306](http://ci.infinispan.org/viewLog.html?buildId=32897&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 9f48bded57500405e54ec0644b943702c56316b2
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:09:17

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 243](http://ci.infinispan.org/viewLog.html?buildId=32857&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 9f48bded57500405e54ec0644b943702c56316b2
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:00:09

Process exited with code 1

**Pull request closed without merging**